node_modules/oae-principals/lib/api.user.js

```diff
@@ -774,3 +811,138 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
         return PrincipalsDAO.setAdmin(adminType, isAdmin, principalId, callback);
     });
 };
+
+/**
+ * Send an email token to a user that can be used to verify the user owns the email address
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {User}       user            The user to send the email token to
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @api private
+ */
+var _sendEmailToken = function(ctx, user, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'userId': user.id}, 'Unable to send a user a verification email');
+        }
+    };
+
+    // Generate a token that will be sent to the user's email address
+    var token = util.format('email-%s', _generateEmailToken());
+
+    // Store the token
+    PrincipalsDAO.storeEmailToken(token, user.id, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // The EmailAPI expects a user to have a verified email address. As this is not the case
+        // when sending an email token, we send in a patched user object
+        var userToEmail = _.extend({}, user, {'email': user.emailChange});
```

Comment:
Maybe it's better to just let the email api take an email address independent of the user profile.